packages/amplify-category-auth/provider-utils/awscloudformation/assets/cognito-defaults.js

```diff
@@ -99,7 +99,9 @@ const entityKeys = {
 };
 
 const getAllDefaults = (name) => {
-  const projectName = name.projectConfig ? `${name.projectConfig.projectName.toLowerCase()}${sharedId}` : name;
+  const disallowedChars = /[^A-Za-z0-9_]+/g;
+  let projectName = name.projectConfig ? `${name.projectConfig.projectName.toLowerCase().substring(0, 100)}${sharedId}` : name.substring(0, 100);
```

Comment:
Why is it 0,100? The max number of chars you can have in a project name is 20.

Reply:
Cognito has limitations of 100 characters. Since we are validating against that this should not hurt